_posts/2023-09-15-kafka-producer-client-essentials.md

```diff
@@ -0,0 +1,372 @@
+---
+layout: post
+title: "Developing Kafka clients: A simple producer application"
+date: 2023-09-15
+author: paul_mellor
+---
+
+Strimzi simplifies the deployment and management of Apache Kafka clusters in a Kubernetes environment, ensuring a smooth and hassle-free experience.
+But the real fun begins once your cluster is up and running.
+That's when you can start thinking about how to interact with Kafka to stream data.
+In other words, Kafka doing the job it excels at.
+
+In this post, we'll dive into the essentials of developing a Kafka producer application that can send messages to a Strimzi-managed Kafka cluster.
+To illustrate these concepts, we'll walk through a basic example of a self-contained application that generates and produces messages to a specific Kafka topic.
+
+> This blog post assumes some basic Java knowledge or experience. 
+> If you're new to Java, familiarizing yourself with the fundamentals of the Java programming language will help you better understand the code and concepts presented here.
+
+## Getting started with your producer application
+
+The first thing to consider when developing a producer application is your preferred programming language. 
+
+After you've decided on that, as a bare minimum, your application must be able to connect to the Kafka cluster and use producers to send messages.
+
+Let's break down the essential steps to get started:
+
+1. Start by choosing the Kafka client library that speaks your choice of programming language. We use Java in this post, but you can use Python, .NET, and so on. The Java client is part of the Apache Kafka project. 
+
+2. Get the library through a package manager or by downloading it from the source.
+
+3. Import the classes and dependencies that your Kafka client will need into your code.
+
+4. Tell your client how to find and connect with your Kafka cluster, specifying an address and port, and, if required, security credentials.
```

Comment:
`specifying an address and port` ... I would say the bootstrap servers list, because in general it's a list of address:port not just one. With Strimzi it's just a service so in the end you will have just one but here we are not mentioning Strimzi yet.